standard/tournaments_summary_table/commons/tournaments_summary_table.lua

```diff
@@ -0,0 +1,305 @@
+			})
+	end
+
+	return conditions
+end
+
+function TournamentsSummaryTable.additionalConditions(type)
+	return {}
+end
+
+function TournamentsSummaryTable.row(eventInformation, type)
+	if type == TournamentsSummaryTable.upcomingType then
+		Variables.varDefine('upcoming_' .. eventInformation.pagename, 1)
+	end
+
+	local displayName = String.isNotEmpty(eventInformation.tickername)
+		and eventInformation.tickername
+		or eventInformation.name
+
+	local icon = ''
+	if not TournamentsSummaryTable.disableLIS then
+		if
+			String.isNotEmpty(eventInformation.series) and
+			TournamentsSummaryTable._lisTemplateExists(eventInformation.series:lower())
+		then
+			icon = eventInformation.series:lower()
+		else
+			icon = 'none'
+		end
+	end
+
+	local iconFile = ''
+	if String.isNotEmpty(eventInformation.icon) then
+		local iconInput = string.gsub(eventInformation.icon, 'File:', '')
+		if mw.title.new('Media:' .. iconInput).exists then
+			iconFile = 'File:' .. iconInput
+		end
+	end
+
+	local iconDarkFile
+	if String.isNotEmpty(eventInformation.icondark) then
+		local iconInput = string.gsub(eventInformation.icondark, 'File:', '')
+		if mw.title.new('Media:' .. iconInput).exists then
+			iconDarkFile = 'File:' .. iconInput
+		end
+	end
+
+	local rowComponents = {
+		'\n** ' .. eventInformation.pagename,
+		displayName,
+		'startdate=' .. TournamentsSummaryTable._dateDisplay(eventInformation.startdate),
+		'enddate=' .. TournamentsSummaryTable._dateDisplay(eventInformation.enddate),
+		'icon=' .. icon,
+		'iconfile=' .. iconFile,
+		'icondarkfile=' .. (iconDarkFile or iconFile),
+	}
+
+	return table.concat(rowComponents, ' | ')
+end
+
+function TournamentsSummaryTable._dateDisplay(dateString)
+	local year, month, day = dateString:match("(%d%d%d%d)-?(%d?%d?)-?(%d?%d?)$")
+	-- fallback
+	if String.isEmpty(year) then
+		return os.time{year=1970, month=1, day=1, hour=0}
+	end
+	-- defaults
+	if String.isEmpty(month) then
+		month = 1
+	end
+	if String.isEmpty(day) then
+		day = 1
+	end
+
+	-- create time
+	local date = os.time{year=year, month=month, day=day, hour=0}
+
+	-- return date display
+	return os.date('%b %d', date)
+end
+
+function TournamentsSummaryTable._lisTemplateExists(series)
+	local lis = Template.safeExpand(
+		mw.getCurrentFrame(),
+		'LeagueIconSmall/' .. series,
+		{},
+		''
+	)
+
+	return String.isNotEmpty(lis)
+end
+
+-- thin wrapper for adding manual upcoming rows (for adding events with unknown dates)
+function TournamentsSummaryTable.manualUpcomingRow(args)
+	args = args or {}
+
+	local pageName = args.pagename or args.link
+	if String.isEmpty(pageName) then
+		error('No pagename specified for manual upcoming row')
+	end
+
+	local eventInformation = {
+		pagename  = pageName,
+		tickername = args.tickername  or args.display or pageName,
+		startdate = string.match(args.estimated_start or args.startdate or '', '%d%d%d%d%-%d%d%-%d%d'),
+		enddate = string.match(args.estimated_end or args.enddate or '', '%d%d%d%d%-%d%d%-%d%d'),
```

Comment:
Entering `2022-07` or `2022-07-??` would fail with this? Use the `_dateDisplay` function?